core/tests/test_utils.py

```diff
@@ -840,6 +841,45 @@ def assertRaisesRegexp(  # pylint: disable=keyword-arg-before-vararg
             expected_exception, expected_regexp,
             callable_obj=callable_obj, *args, **kwargs)
 
+    def assert_matches_regexps(self, items, regexps):
+        """Asserts that each item is a full match of the corresponding regexp.
+
+        Every regexp in the list must be a full match, substring matches are
+        still treated as errors.
+
+        If there are any missing or extra items that do not correspond to a
+        regexp element, then that is also an error.
+
+        Args:
+            items: list(str). The string elements being matched.
+            regexps: list(str|RegexObject). The patterns that each item is
+                expected to match.
+
+        Raises:
+            AssertionError. At least one item does not match its corresponding
+                pattern, or the number of items does not match the number of
+                regexp patterns.
+        """
+        differences = [
+            '~ [i=%d]:\t%r does not match: %r' % (i, item, regexp)
+            for i, (regexp, item) in enumerate(python_utils.ZIP(regexps, items))
+            if re.match(regexp, item) is None
+        ]
+        if len(items) < len(regexps):
+            extra_regexps = regexps[len(items):]
+            differences.extend(
```

Comment:
Shouldn't these also raise an exception? From what I understand, ZIP would ignore all extra elements. 

In the case where the sizes are different, but there are no differences in the list, the check would pass right?

Reply:
Oh oops, sorry I got a bit confused... ignore this comment.

Reply:
That's no problem, I'd actually prefer cleaning it up so that there's less confusion.

Do you think I can change the code to make it more obvious?

Reply:
Just a note though, if the follow-up changes don't affect functionality, can they be done in a separate PR? We really need to get this one in soonish.

Reply:
Yup of course!